### PR TITLE
Core/Spells: Fixed Beacon of Light not healing pet(s)

### LINKED ISSUE #15609: [3.3.5a] Beacon of Light is not healing pet
Beacon of Light is not healing pet.

Revision: 747a3f1d38abdb9b711877c168243f0d398b615f

### PR DESCRIPTION
**Pull request by Goatform**
Closes #15609

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -6382,27 +6382,28 @@ bool Unit::HandleDummyAuraProc(Unit* victim, uint32 damage, AuraEffect* triggere
                 }
                 else
                 {    // Check Party/Raid Group
-                    if (Group* group = ToPlayer()->GetGroup())
```
Comment (jackpoz) [inline src/server/game/Entities/Unit/Unit.cpp]: did the old code trigger a NULL dereference exception at this point when IsPet() returned true ?

Comment (jackpoz) [inline src/server/game/Entities/Unit/Unit.cpp]: Actually, how can IsPet() return true here inside the else of https://github.com/Goatform/TrinityCore/blob/3.3.5/src/server/game/Entities/Unit/Unit.cpp#L6375 check ? Was this change tested ingame AND debugged with both cases IsPet() true/false ?

Comment (Goatform) [inline src/server/game/Entities/Unit/Unit.cpp]: I tested it and I was using this fix for a long time and none found any issue related to it.

Comment (jackpoz) [inline src/server/game/Entities/Unit/Unit.cpp]: I wrote above what's wrong with this code, there is dead code in it https://en.wikipedia.org/wiki/Dead_code

File: src/server/game/Entities/Unit/Unit.cpp [hunk 2]
```diff
@@ -6382,27 +6382,28 @@ bool Unit::HandleDummyAuraProc(Unit* victim, uint32 damage, AuraEffect* triggere
...
-                    {
+                    if (Group* group = IsPet() ? GetOwner()->ToPlayer()->GetGroup() : ToPlayer()->GetGroup())
                         for (GroupReference* itr = group->GetFirstMember(); itr != NULL; itr = itr->next())
-                        {
```
Comment (Treeston) [inline src/server/game/Entities/Unit/Unit.cpp]: Code style says brackets should be used whenever there is more than one line in the code block.

File: src/server/game/Entities/Unit/Unit.cpp (hunk 1 above)
Comment (Treeston) [inline src/server/game/Entities/Unit/Unit.cpp]: @Goatform The line will never be reached for pets, there is a check for typeid being `TYPEID_PLAYER` further up.

**Comment by Goatform** [comment]
@jackpoz & @Treeston check it now.

**Comment by jackpoz** [comment]
It looks better now

**Comment by joschiwald** [comment]
disagree, whole beacon of light code is ugly

**Comment by cemak** [comment]
check guardian too ?

**Pull request closed without merging**